### PR TITLE
[P1-N06] Small formatting corrections to DVM

### PR DESCRIPTION
**Pull request by chrismaree**
This PR cleans up a number of small formatting changes throughout the DVM's comments to make things more consistent. Specifically this PR introduces a styling convention for dealing with internal and private variables, functions and underscores. The styling rules are now as follows:

- Private/Internal functions all proceed with underscore in all solidity EXCEPT for libraries. This applies to all `contract` or `abstract contract`.
- private variables do not have underscores in any context.
- parameters never have underscores except in constructors. This rule because often a constructor is assigning a lot of variables where the internal representation is the same name as the parm and this helps with clashing variable names.

As a later issue we should update the `STYLE.md` for the repo to reflect our opinions on styling as they evolve. I've opened an issue to keep track of this [here](https://github.com/UMAprotocol/protocol/issues/1277).

### CODE REVIEW

File: core/contracts/oracle/implementation/DesignatedVoting.sol [hunk 1]
```diff
@@ -88,9 +87,10 @@ contract DesignatedVoting is Withdrawable {
 
     /**
      * @notice Forwards a reward retrieval to Voting.
-     * @dev rewards are added to the tokens already held by this contract.
+     * @dev Rewards are added to the tokens already held by this contract.
      * @param roundId defines the round from which voting rewards will be retrieved from.
      * @param toRetrieve an array of PendingRequests which rewards are retrieved from.
+     * @return totalRewardToIssue total amount of rewards returned to the voter.
```
Comment (nicholaspai) [inline core/contracts/oracle/implementation/DesignatedVoting.sol:93]: @chrismaree I addressed this NatSpec inconsistency in #1270

Comment (chrismaree) [reply core/contracts/oracle/implementation/DesignatedVoting.sol:93]: great.

**Comment by chrismaree** [comment]
> What about the Withdrawable functions that don't start with `_`?

This is a good point and is something we need to agree upon for a styling convention regarding `_`s in general. In some contracts we preceed internal variables with underscores and in others we proceed internal functions with underscores. We need to have a standard rule here. I propose the following:

- Private/Internal functions all proceed with underscore.
- private variables do not have underscores.
- parameters never have underscores except in constructors. This rule because often a constructor is assigning a lot of variables where the internal representation is the same name as the parm.

I make these suggestions because:
1) they closely match what we have now (only a few small changes) which means small lower churn. All three rules are followed in almost all contracts already. 
2) they are light weight on formatting.

@mrice32 & @ptare @nicholaspai what do you think? if you like this I can implement it in the other contracts across the board.

**Comment by mrice32** [comment]
> > What about the Withdrawable functions that don't start with `_`?
> 
> This is a good point and is something we need to agree upon for a styling convention regarding `_`s in general. In some contracts we preceed internal variables with underscores and in others we proceed internal functions with underscores. We need to have a standard rule here. I propose the following:
> 
> * Private/Internal functions all proceed with underscore.
> * private variables do not have underscores.
> * parameters never have underscores except in constructors. This rule because often a constructor is assigning a lot of variables where the internal representation is the same name as the parm.
> 
> I make these suggestions because:
> 
> 1. they closely match what we have now (only a few small changes) which means small lower churn. All three rules are followed in almost all contracts already.
> 2. they are light weight on formatting.
> 
> @mrice32 & @ptare @nicholaspai what do you think? if you like this I can implement it in the other contracts across the board.

SGTM, but I don't want to create too much churn or waste a lot of time trying to get perfect consistency.  IMO, if we think getting this consistency is doable with a fairly small addition to this PR, let's do it. Otherwise, we can just not fix N06.

**Comment by nicholaspai** [comment]
> > What about the Withdrawable functions that don't start with `_`?
> 
> This is a good point and is something we need to agree upon for a styling convention regarding `_`s in general. In some contracts we preceed internal variables with underscores and in others we proceed internal functions with underscores. We need to have a standard rule here. I propose the following:
> 
> * Private/Internal functions all proceed with underscore.
> * private variables do not have underscores.
> * parameters never have underscores except in constructors. This rule because often a constructor is assigning a lot of variables where the internal representation is the same name as the parm.
> 
> I make these suggestions because:
> 
> 1. they closely match what we have now (only a few small changes) which means small lower churn. All three rules are followed in almost all contracts already.
> 2. they are light weight on formatting.
> 
> @mrice32 & @ptare @nicholaspai what do you think? if you like this I can implement it in the other contracts across the board.

I agree with this standard

**Comment by nicholaspai** [comment]
> > > What about the Withdrawable functions that don't start with `_`?
> > 
> > 
> > This is a good point and is something we need to agree upon for a styling convention regarding `_`s in general. In some contracts we preceed internal variables with underscores and in others we proceed internal functions with underscores. We need to have a standard rule here. I propose the following:
> > 
> > * Private/Internal functions all proceed with underscore.
> > * private variables do not have underscores.
> > * parameters never have underscores except in constructors. This rule because often a constructor is assigning a lot of variables where the internal representation is the same name as the parm.
> > 
> > I make these suggestions because:
> > 
> > 1. they closely match what we have now (only a few small changes) which means small lower churn. All three rules are followed in almost all contracts already.
> > 2. they are light weight on formatting.
> > 
> > @mrice32 & @ptare @nicholaspai what do you think? if you like this I can implement it in the other contracts across the board.
> 
> SGTM, but I don't want to create too much churn or waste a lot of time trying to get perfect consistency. IMO, if we think getting this consistency is doable with a fairly small addition to this PR, let's do it. Otherwise, we can just not fix N06.

I think Chris' proposal is a good tradeoff between consistency and not having to make big changes (such as making all parameters start with an underscore)

File: core/contracts/common/implementation/ExpandedERC20.sol [hunk 2]
```diff
@@ -21,15 +21,15 @@ contract ExpandedERC20 is ExpandedIERC20, ERC20, MultiRole {
...
      * @notice Constructs the ExpandedERC20.
-     * @param tokenName The name which describes the new token.
-     * @param tokenSymbol The ticker abbreviation of the name. Ideally < 5 chars.
-     * @param tokenDecimals The number of decimals to define token precision.
+     * @param _tokenName The name which describes the new token.
+     * @param _tokenSymbol The ticker abbreviation of the name. Ideally < 5 chars.
+     * @param _tokenDecimals The number of decimals to define token precision.
      */
-    constructor(string memory tokenName, string memory tokenSymbol, uint8 tokenDecimals)
+    constructor(string memory _tokenName, string memory _tokenSymbol, uint8 _tokenDecimals)
```
Comment (chrismaree) [inline core/contracts/common/implementation/ExpandedERC20.sol:28]: All constructors must have underscores.

File: core/contracts/common/implementation/Withdrawable.sol [hunk 3]
```diff
@@ -17,39 +17,39 @@ import "./MultiRole.sol";
...
     /**
      * @notice Internal method that allows derived contracts to create a role for withdrawal.
-     * @dev Either this method or `setWithdrawRole` must be called by the derived class for this contract to function
+     * @dev Either this method or `_setWithdrawRole` must be called by the derived class for this contract to function
      * properly.
      */
-    function createWithdrawRole(uint256 roleId, uint256 managingRoleId, address withdrawerAddress) internal {
-        _roleId = roleId;
-        _createExclusiveRole(roleId, managingRoleId, withdrawerAddress);
+    function _createWithdrawRole(uint256 newRoleId, uint256 managingRoleId, address withdrawerAddress) internal {
```
Comment (chrismaree) [inline core/contracts/common/implementation/Withdrawable.sol:42]: this param had to change to not clash with the local `roleId`

File: core/contracts/common/implementation/Testable.sol [hunk 4]
```diff
@@ -31,10 +31,10 @@ abstract contract Testable {
     /**
      * @notice Sets the current time.
      * @dev Will revert if not running in test mode.
-     * @param _time timestamp to set current Tesable time to.
+     * @param time timestamp to set current Tesable time to.
      */
-    function setCurrentTime(uint _time) external onlyIfTest {
-        Timer(timerAddress).setCurrentTime(_time);
+    function setCurrentTime(uint time) external onlyIfTest {
+        Timer(timerAddress).setCurrentTime(time);
```
Comment (mrice32) [inline core/contracts/common/implementation/Testable.sol:37]: Why were these changed to *not* have preceding underscores but the previous file was changed to add preceding underscores to function args?

File: core/contracts/common/implementation/Timer.sol [hunk 5]
```diff
@@ -14,10 +14,10 @@ contract Timer {
     /**
      * @notice Sets the current time.
      * @dev Will revert if not running in test mode.
-     * @param _time timestamp to set current Tesable time to.
+     * @param time timestamp to set current Tesable time to.
      */
-    function setCurrentTime(uint _time) external {
-        currentTime = _time;
+    function setCurrentTime(uint time) external {
+        currentTime = time;
```
Comment (mrice32) [inline core/contracts/common/implementation/Timer.sol:20]: Same as above

File: core/contracts/oracle/implementation/DesignatedVoting.sol [hunk 6]
```diff
@@ -37,7 +36,7 @@ contract DesignatedVoting is Withdrawable {
     constructor(address finderAddress, address ownerAddress, address voterAddress) public {
         _createExclusiveRole(uint(Roles.Owner), uint(Roles.Owner), ownerAddress);
         _createExclusiveRole(uint(Roles.Voter), uint(Roles.Owner), voterAddress);
-        setWithdrawRole(uint(Roles.Owner));
+        _setWithdrawRole(uint(Roles.Owner));
```
Comment (chrismaree) [inline core/contracts/oracle/implementation/DesignatedVoting.sol:39]: This is now consistent with the other two calls to `_createExclusiveRole` because `Withdrawable.sol` is abstract and so should be preceeded by an underscore for internal & private functions.

File: core/contracts/common/implementation/Testable.sol (hunk 4 above)
Comment (nicholaspai) [reply core/contracts/common/implementation/Testable.sol:37]: I think this is consistent with the rule "parameters never have underscores except in constructors."

Comment (mrice32) [reply core/contracts/common/implementation/Testable.sol:37]: Ahh, I see. Ignore the above. I got my wires crossed. Thought this was a constructor.

File: core/contracts/common/implementation/Timer.sol (hunk 5 above)
Comment (mrice32) [reply core/contracts/common/implementation/Timer.sol]: Same as above -- ignore this comment.

Comment (chrismaree) [reply core/contracts/common/implementation/Timer.sol]: external non-constructor function.

File: core/contracts/common/implementation/Testable.sol (hunk 4 above)
Comment (chrismaree) [reply core/contracts/common/implementation/Testable.sol:37]: yup, that's right @nicholaspai

**Comment by mrice32** [comment]
@chrismaree looks like ci is complaining about some natspec issue.

**Comment by chrismaree** [comment]
> @chrismaree looks like ci is complaining about some natspec issue.

Yup. on it!
